test/library/metrics/query.go

```diff
@@ -0,0 +1,107 @@
+package metrics
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	routeclient "github.com/openshift/client-go/route/clientset/versioned"
+	"github.com/prometheus/common/model"
+)
+
+type PrometheusClient struct {
```

Comment:
i would make this private struct and return `interface` in NewPrometheusClient below... It allows mocking or replacing the client in future. Also a public struct with all fields private is strange to me.

Reply:
_still working on this_

Reply:
I exported both struct and fields - I'm confused about returning the interface, using private struct.  In Query fn, we need the Host+Token so shouldn't the struct + fields be exported?  Maybe it'll make more sense if i stare at it a bit more.

Reply:
now that I'm using prometheus/client_golang this is resolved.